utils/jsonmerge/jsonmerge.bal

```diff
@@ -0,0 +1,284 @@
+import ballerina/log;
+import ballerina/regex;
+
+// Configuration flag to control behavior when source and base have different types for the same key
+// When true: ignores type mismatches and preserves base value
+// When false: returns an error on type mismatch
+configurable boolean ignoreMismatchedTypes = false;
+
+# Description.
+#
+# + srcJson - Source JSON object to merge from
+# + baseJson - Base JSON object to merge into (serves as base)  
+# + keys - Optional map specifying merge keys for arrays at different paths
+# Format: {"arrayFieldName": "keyField"} or {"parent.arrayField": "keyField"}
+# Supports composite keys: {"arrayField": "key1,key2"}
+# Note: Currently only supports keys with primitive values (string, int, boolean, float, decimal)
+# + return - return merged JSON object or error if types mismatch
+public isolated function mergeJson(json baseJson, json srcJson, map<string>? keys = ()) returns json|error {
```

Comment:
better to have "original" instead of "baseJson" and "updates" instead of "srcJson"